src/bin/busd.rs

```diff
@@ -51,10 +66,30 @@ impl From<AuthMechanism> for zbus::AuthMechanism {
 async fn main() -> Result<()> {
     busd::tracing_subscriber::init();
 
-    let args = Args::parse();
+    let mut args = BusdArgs::parse();
+
+    let mut config = BusConfig::default();
+    if let Some(file) = args.config.config_file {
+        config = BusConfig::read(file)?;
+    }
+
+    if let Some(address) = args.address {
+        config.elements.push(config::Element::Listen(address));
+    }
+
+    // FIXME: we don't support multiple <listen> atm
+    let address = config
+        .elements
+        .iter()
+        .rev()
+        .find_map(|e| match e {
+            config::Element::Listen(l) => Some(l),
+            _ => None,
+        })
+        .unwrap();
```

Comment:
I think `BusConfig` should provide specific API for each element so user doesn't have to deal with generic elements. It's exposing the detail that config is xml-based IMO.